### PR TITLE
[PVR] Timerdialog fixes

### PR DESCRIPTION
**Pull request by Glenn-1990**
The label for max recordings was: "Max recordings: End any time %d"..
now -> "Max recordings: %d"

When the backend sends an non standard integer value for lifetime, priority or max recordings, the setting stays empty. We should add the missing value instead (like start and end margin)
The pvr add-on can not add all the possible integer values to the valueslist as it would be huge, that's why we should handle this.

@ksooo @Jalle19

### CODE REVIEW

**Comment by ksooo** [comment]
Despite from nitpicking re bool var, +1 from me.

**Comment by Glenn-1990** [comment]
@ksooo removed the boolean ;-)

File: xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp [hunk 1]
```diff
@@ -923,6 +923,21 @@ void CGUIDialogPVRTimerSettings::PrioritiesFiller(
     list.clear();
     pThis->m_timerType->GetPriorityValues(list);
     current = pThis->m_iPriority;
+
+    auto it = list.begin();
```
Comment (Jalle19) [inline xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp]: `auto it = std::find(list.begin(), list.end(), current);`

File: xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp [hunk 2]
```diff
@@ -937,6 +952,21 @@ void CGUIDialogPVRTimerSettings::LifetimesFiller(
     list.clear();
     pThis->m_timerType->GetLifetimeValues(list);
     current = pThis->m_iLifetime;
+
+    auto it = list.begin();
```
Comment (Jalle19) [inline xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp]: Same here obviously.

File: xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp [hunk 3]
```diff
@@ -951,6 +981,21 @@ void CGUIDialogPVRTimerSettings::MaxRecordingsFiller(
     list.clear();
     pThis->m_timerType->GetMaxRecordingsValues(list);
     current = pThis->m_iMaxRecordings;
+
+    auto it = list.begin();
```
Comment (Jalle19) [inline xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp]: And here.

File: xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp (hunk 1 above)
Comment (Glenn-1990) [inline xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp]: but we need to check 'it->second', can this be done with std::find?

Comment (Jalle19) [inline xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp]: You're right, you'd have to use `std::find_if`.

```
auto it = std::find_if(list.begin(), list.end(), [current](std::pair<foo, bar> &mapEntry) {
    return mapEntry.second == current;
}
```

Up to which way to do it, the current approach is okay too.

**Comment by ksooo** [comment]
jenkins build this please

**Pull request merged**